### PR TITLE
Add ability to download JDKs from Oracle using auth info from Databag.

### PR DESCRIPTION
**Pull request by cbajumpaa**

### CODE REVIEW

**Comment by erichelgeson** [comment]
Hi there,

I get what you're trying to accomplish here, but if you could make it generic (no need to reference datadog, not sure how that applies) - remove encrypted databags and use attributes (you could pass in them from a wrapper script instead of using the directly here) and endure it works without auth as well (since it does work with at least 1.8 currently)

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license           'Apache 2.0'
 description       'Installs Java runtime.'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '1.50.0'
+version           '1.50.1'
```
Comment (erichelgeson) [inline metadata.rb]: No need to bump versions, I will do this in a release process.

File: providers/ark.rb [hunk 2]
```diff
@@ -79,10 +79,23 @@ def download_direct_from_oracle(tarball_name, new_resource)
...
-      shell_out!(
-        %( curl --create-dirs -L --retry #{new_resource.retries} --retry-delay #{new_resource.retry_delay} --cookie "#{cookie}" #{new_resource.url} -o #{download_path} --connect-timeout #{new_resource.connect_timeout} #{proxy} ),
-                                 timeout: new_resource.download_timeout
-      )
+      if new_resource.oracle_username && new_resource.oracle_password
+        Chef::Log.debug 'Using auth info from databag'
+        username = new_resource.oracle_username
+        password = new_resource.oracle_password
+        cmd = shell_out!(<<-EOH
+          SIGNINURL=$(curl --retry #{new_resource.retries} --retry-delay #{new_resource.retry_delay} --connect-timeout #{new_resource.connect_timeout} -c /tmp/cookies.txt -o /dev/null -v http://www.oracle.com/webapps/redirect/signon?nexturl=#{new_resource.url} #{proxy} 2>&1 | grep "< Location: " | awk '{print $3}' | tr -d '\r' );
```
Comment (erichelgeson) [inline providers/ark.rb]: Wouldn't `-L` follow redirect do the same thing here without 2 commands?

File: recipes/oracle.rb [hunk 3]
```diff
@@ -58,6 +58,20 @@
...
 end
 
+username = nil
+password = nil
+
+if node['java']['oracle_user_databag_item'] &&
+   node['java']['oracle_user_databag'] &&
+   node['java']['oracle_user_databag_secret']
+
+  secret = Chef::EncryptedDataBagItem.load_secret(node['java']['oracle_user_databag_secret'])
```
Comment (erichelgeson) [inline recipes/oracle.rb]: any secret management should be done outside of this cookbook and not reference a 3rd party company or integration you are using (datadog)

**Comment by cbajumpaa** [comment]
> you could pass in them from a wrapper script instead of using the directly here

Do you have an example of this?

**Comment by erichelgeson** [comment]
I would let the user of the cookbook set the attributes
```
node['java']['oracle_user_databag']
node['java']['oracle_user_databag_secret']
```
Since some users may use databags, or other means to store their secrets.

**Pull request closed without merging**